akka-actor-typed-tests/src/test/java/jdocs/akka/typed/supervision/SupervisionCompileOnlyTest.java

```diff
@@ -135,4 +133,33 @@ public void supervision() {
   }
   // #restart-keep-children
 
+  interface Resource {
+    void close();
+    void process(String[] parts);
+  }
+  public static Resource claimResource() {
+    return null;
+  }
+
+  static void prerestartBehavior() {
+    //#restart-PreRestart-signal
+    Behaviors.supervise(
+      Behaviors.<String>setup(ctx -> {
+        final Resource resource = claimResource();
+
+        return Behaviors.receive((notUsed, msg) -> {
```

Comment:
receiveMessage

Reply:
ah, it's because of signal handler. Better to use the builder:
```java
    return Behaviors.receive(String.class)
          .onMessage(String.class, msg -> {
            // there might be bugs here...
            String[] parts = msg.split(" ");
            resource.process(parts);
            return Behaviors.same();
          })
          .onSignal(PreRestart.class, signal -> {
            resource.close();
            return Behaviors.same();
          })
          .onSignal(PostStop.class, signal -> {
            resource.close();
            return Behaviors.same();
          })
          .build();
```

Reply:
Yeah, felt silly to do that when the top type is the only message (`String`) but this looks better.